scripts/clean_osm_data.py

```diff
@@ -279,279 +288,320 @@ def finalize_lines_type(df_lines):
-]
-dropped_circuits_tags = [
-    x for x in circuits_tag_to_n_circuits.keys() if circuits_tag_to_n_circuits[x] == "0"
-]
+    # replace raw values
+    repl_freq = {
+        "16.67": "16.7",
+        "50;50;16.716.7": "50;50;16.7;16.7",
+        "50;50;16.716.7": "50;50;16.7;16.7",
+        "50;16.7?": "50;16.7",
+        # "24 kHz": "24000",
+    }
 
+    # TODO: default frequency may be by country
+    # default frequency
+    default_frequency = "50"
 
-def integrate_lines_df(df_all_lines, distance_crs):
+    df["tag_frequency"] = (
+        df["tag_frequency"].replace(repl_freq).fillna(default_frequency)
+    )
+
+    return df
+
+
+def parse_voltage(df):
+    "Function to parse raw voltage column: manual fixing and drop undesired values"
+
+    # replace raw values
+    repl_voltage = {
+        "medium": "33000",
+        "19.1 kV": "19100",
+        "high": "220000",
+        "240 VAC": "240",
+        "2*220000": "220000;220000",
+        "KV30": "30kV",
+    }
+
+    df["voltage"] = (
+        df["voltage"]
+        .replace(repl_voltage)
+        .str.lower()
+        .str.replace(" ", "")
+        .str.replace("_", "")
+        .str.replace("kv", "000")
+        .str.replace("v", "")
+        # .str.replace("/", ";")
+    )
+
+    # drop na values in voltage
+    df.dropna(subset=["voltage"], inplace=True)
+
+    return df
+
+
+def parse_circuits(df):
+    "Function to parse raw circuits column: manual fixing and drop undesired values"
+
+    # replace raw values
+    repl_circuits = {
+        "1/3": "0",
+        "2/3": "0",
+        # assumption of two lines and one grounding wire
+        "2-1": "2",
+        "single": "1",
+        "partial": "1",
+        "1;1 disused": "1;0",
+        # assuming that in case of a typo there is at least one line
+        "`": "1",
+        "^1": "1",
+        "e": "1",
+        "d": "1",
+        "1.": "1",
+    }
+
+    df["circuits"] = df["circuits"].replace(repl_circuits).str.replace(" ", "")
+
+    return df
+
+
+def parse_cables(df):
+    "Function to parse raw cables column: manual fixing and drop undesired values"
+
+    # replace raw values
+    repl_cables = {
+        "single": "1",
+        "triple": "3",
+        "partial": "1",
+        "1 disused": "0",
+        "3;3 disused": "3;0",
+        "1 (Looped - Haul & Return) + 1 power wire": "1",
+        "ground": "0",
+        "line": "1",
+        # assuming that in case of a typo there is at least one line
+        "`": "1",
+        "^1": "1",
+        "e": "1",
+        "d": "1",
+        "2-1": "3",
+        "3+3": "6",
+        "6+1": "6",
+        "2x3": "6",
+        "3x2": "6",
+        "2x2": "4",
+    }
+
+    df["cables"] = df["cables"].replace(repl_cables).str.replace(" ", "")
+
+    return df
+
+
+def split_and_match_vf(df):
     """
-    Function to add underground, under_construction, frequency and circuits
+    Function to match the length of the columns in subset
```

Comment:
vf stands for voltage and frequency?

Reply:
Docstring is not clear. It talk about match but what about the split? Why does it exist? Regarding the 'match', see two comments further down what I suggest. Can you adapt it here as well?

Just checking the full function, this looks fairly complex

Reply:
> vf stands for voltage and frequency?

Yes, revised

Reply:
> Docstring is not clear. It talk about match but what about the split? Why does it exist? Regarding the 'match', see two comments further down what I suggest. Can you adapt it here as well?
> 
> Just checking the full function, this looks fairly complex

Docstring revised; the algorithm may be changed to be more conservative in case.
We can talk about it